pilot/pkg/networking/core/v1alpha3/tracing.go

```diff
@@ -315,8 +317,37 @@ func convert(ctxs []meshconfig.MeshConfig_ExtensionProvider_OpenCensusAgentTraci
 	return converted
 }
 
+func dryRunPolicyTraceTag(name, key string) *tracing.CustomTag {
+	return &tracing.CustomTag{
+		Tag: name,
+		Type: &tracing.CustomTag_Metadata_{
+			Metadata: &tracing.CustomTag_Metadata{
+				Kind: &envoy_type_metadata_v3.MetadataKind{
+					Kind: &envoy_type_metadata_v3.MetadataKind_Request_{
+						Request: &envoy_type_metadata_v3.MetadataKind_Request{},
+					},
+				},
+				MetadataKey: &envoy_type_metadata_v3.MetadataKey{
+					Key: authz_model.RBACHTTPFilterName,
+					Path: []*envoy_type_metadata_v3.MetadataKey_PathSegment{
+						{
+							Segment: &envoy_type_metadata_v3.MetadataKey_PathSegment_Key{
+								Key: key,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
 func defaultTags() []*tracing.CustomTag {
 	return []*tracing.CustomTag{
+		dryRunPolicyTraceTag("istio.authorization.dry_run.allow_policy.name", authz_model.RBACShadowRulesAllowStatPrefix+authz_model.RBACShadowEffectivePolicyID),
```

Comment:
how will this impact users that don't use this feature?

Reply:
this controls the custom tag showing in the trace span, if the user has no authz policy in dry-run, these tags will not be showed (no-op).

Reply:
Not shown in a UI or have no impact at all? is there storage cost, etc of these? I recall some flags in the past to limit tags so it seems important?

Reply:
there shouldn't be any storage cost, the tag just does not exist. there will be slightly overhead for trying to get the value for the tag from the Envoy request metadata (it does not exist in this case).